tests/lib/rules/key-spacing.js

```diff
@@ -313,6 +313,81 @@ ruleTester.run("key-spacing", rule, {
             "align": "value",
             "beforeColon": true
         }]
+    },
+
+    // https://github.com/eslint/eslint/issues/4763
+    {
+        code: "({a : foo, ...x, b : bar})['a'];",
+        options: [{
+            beforeColon: true,
+            afterColon: true
+        }],
+        parserOptions: { ecmaVersion: 6, ecmaFeatures: { experimentalObjectRestSpread: true } }
+    }, {
+        code: [
+            "var obj = {",
+            "    'a'     : (42 - 12),",
+            "    ...x,",
+            "    foobar  : 'value',",
+            "    [(expr)]: val",
+            "};"
+        ].join("\n"),
+        options: [{
+            align: "colon"
+        }],
+        parserOptions: { ecmaVersion: 6, ecmaFeatures: { experimentalObjectRestSpread: true } }
+    }, {
+        code: [
+            "callExpr(arg, {",
+            "    key       :val,",
+            "    ...x,",
+            "    ...y,",
+            "    'another' :false,",
+            "    [compute] :'value'",
+            "});"
+        ].join("\n"),
+        options: [{
+            align: "colon",
+            beforeColon: true,
+            afterColon: false
+        }],
+        parserOptions: { ecmaVersion: 6, ecmaFeatures: { experimentalObjectRestSpread: true } }
+    }, {
+        code: [
+            "var obj = {",
+            "    a:        (42 - 12),",
+            "    ...x,",
+            "    'foobar': 'value',",
+            "    bat:      function() {",
+            "        return this.a;",
+            "    },",
+            "    baz: 42",
+            "};"
+        ].join("\n"),
+        options: [{
+            align: "value"
+        }],
+        parserOptions: { ecmaVersion: 6, ecmaFeatures: { experimentalObjectRestSpread: true } }
+    }, {
+        code: [
+            "({",
+            "    ...x,",
+            "    a  : 0,",
+            "    // same group",
+            "    bcd: 0, /*",
+            "    end of group */",
+            "",
+            "    // different group",
+            "    e: 0,",
+            "    ...y,",
+            "    /* group b */",
+            "    f: 0",
+            "})"
+        ].join("\n"),
+        options: [{
+            align: "colon"
+        }],
+        parserOptions: { ecmaVersion: 6, ecmaFeatures: { experimentalObjectRestSpread: true } }
```

Comment:
Can you add an invalid test as well to verify that it will warn if a normal property that comes after a spread property in a group doesn't match the alignment of the rest of the group before the spread property?